### ISSUE TITLE
tar 2.0.0 fails to build where 1.1.0 builds fine

### ISSUE THREAD

**Issue by vog** [mirage/ocaml-tar#85]
I have a local OPAM switch into which I can't install tar-unix 2.0.0 as it already fails when building the dependency tar 2.0.0.

OPAM error message:

```
# context     2.1.2 | linux/x86_64 | ocaml-option-flambda.1 ocaml-option-nnpchecker.1 ocaml-variants.4.13.1+options | https://opam.ocaml.org#9a1bb2dc
# path        ~/.../_opam/.opam-switch/build/tar.2.0.0
# command     ~/.opam/opam-init/hooks/sandbox.sh build dune install -p tar --create-install-files tar
# exit-code   1
# env-file    ~/.opam/log/tar-20092-fb936e.env
# output-file ~/.opam/log/tar-20092-fb936e.out
### output ###
# Error: Program opam not found in the tree or in PATH
```

Note that my OPAM binary is indeed not in PATH, but even if I add it the PATH, I just get a different error message:

```
...
### output ###
# /.../_opam/bin/opam var prefix > /opam-tmp/dune_db4930_output
# [ERROR] Opam has not been initialised, please run `opam init'
```

I see two issues here:
* First, the sandbox seems to exclude ~/.opam and hence prevents calls to opam from working.
  * This might actually be a bug in `dune` rather than `tar`.
* Second, even if the first problem was fixed, `opam` is called in a very dangerous way!
  * It is false to assume that the currently running OPAM instance is the first hit in the path.
  * Typical counter example: You have "opam" in the PATH from your Linux distro, but you prefer a different (perhaps newer) version for your specific project. So you download opam directly into the project and call that binary for further steps.
  * So either you should find a way to determine that actually running OPAM instance, or you should not be doing sub-calls to OPAM in the place.

My current workaround is:

```
tar-unix!=2.0.0
```

which currently installs tar-unix 1.1.0 with tar 1.1.0 and everything is fine.

In case it is important, my OPAM version is 2.1.2 and my local switch is configured as: ocaml-variants.4.13.1+options,ocaml-option-flambda,ocaml-option-nnpchecker.

**Comment by MisterDA** [comment]
Hi @vog, tar and tar-unix are quite standard opam packages, really not doing at all anything out of the ordinary with the build instructions. I don't think any of the two issues you've listed apply directly to tar or tar-unix. I've successfully build latest ocaml-tar with the switch you're using.

```
ocaml-tar $ opam switch create -y . --packages=ocaml-variants.4.13.1+options,ocaml-option-flambda,ocaml-option-nnpchecker
```

I suggest you take a look at your opam install, the shell script hooks to set up the correct opam environment and detect local switches may not be properly set up. Try running `eval $(opam env)` before building ocaml-tar.

**Comment by hannesm** [comment]
To me this looks very similar to what got reported in https://github.com/ocaml/dune/issues/5455 -- would you mind to use dune 2.x instead of dune 3.x and check whether tar 2.0.0 works then fine on your system?

**Comment by vog** [comment]
Yes, you are right. I can confirm that specifying `'dune<3.0.0'` in the package list, my project builds perfectly fine with tar-unix 2.0.1.

So this is actually a bug in dune (see https://github.com/ocaml/dune/issues/5455), not a bug in the tar package.
